File: solutions/java/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
        if (n == 0) return 0;
        if (n <= 2) return 1;
        int a = 0, b = 1, c = 1;
        for (int i = 0; i < n - 2; i++) {
            int temp = a + b + c;
            a = b;
            b = c;
            c = temp;
        }
        return c;
    }
}

File: solutions/python/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0: 
            return 0 
        elif n <= 2: 
            return 1 
        a, b, c = 0, 1, 1 
        for _ in range(n - 2): 
            a, b, c = b, c, a + b + c 
        return c 
                                                                                        
